utils/python/CIME/XML/namelist_defaults.py

```diff
@@ -0,0 +1,113 @@
+"""Interface to `namelist_defaults.xml`.
```

Comment:
I don't think that `entry_id` supports any kind of ranking system for selecting among multiple matches (?), so the search code here is not really duplicating any existing `entry_id` functionality. Even after this is redesigned, we still need some way for specific matches to override non-specific matches, which will be doing something similar to this module.